### PR TITLE
Multilabel classification

### PR DESCRIPTION
**Pull request by sovrasov**

### CODE REVIEW

File: torchreid/losses/asl.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            one_sided_gamma = self.gamma_pos * targets + self.gamma_neg * (1 - targets)
+            one_sided_w = torch.pow(1 - pt, one_sided_gamma)
+            if self.disable_torch_grad_focal_loss:
+                torch.set_grad_enabled(True)
+            loss *= one_sided_w
+
+        return -loss.sum()
+
+
+class AsymmetricLossOptimized(nn.Module):
```
Comment (kprokofi) [inline torchreid/losses/asl.py]: I think we need to keep one of the loss implementations. Either this or one above. I vote for keeping this one, it's optimizing GPU uploading a bit.

File: torchreid/engine/engine.py [hunk 2]
```diff
@@ -623,8 +624,18 @@ def test(
                             ranks=ranks,
                             lr_finder = lr_finder
                         )
-                elif get_model_attr(model, 'contrastive'):
+                elif model_type == 'contrastive':
                     pass
+                elif model_type == 'multilabel':
```
Comment (kprokofi) [inline torchreid/engine/engine.py:629]: Need to add EMA supporting for multilabel classification too. I will do it as it is on my thread.

**Comment by kprokofi** [comment]
@sovrasov , please, check my commits. If you agree - we can merge PR.
